tests/test_subparser.py

```diff
@@ -251,12 +251,12 @@ def test_flag(self, capsys):
         assert "unrecognized arguments: --flag" in captured.err
 
     def test_command_args(self):
-        config = parse(self.Config, ["Command1", "--cmd-a", "12"])
+        config = parse(self.Config, ["Command1", "--a", "12"])
```

Comment:
I think it would be nice to add a property `a: int = 10` to Config, and another test case:

```python
    def test_command_args_collision(self):
        config = parse(self.Config, ["--a", "10", "Command1", "--a", "12"])
        assert config.a == 10
        assert config.cmd.sub_command.a == 12
        assert config.flag is False
```

to showcase the non-prefixed "--a" after the command is parsed correctly